compiler/qsc_doc_gen/src/display.rs

```diff
@@ -335,25 +345,105 @@ struct IdentTyDef<'a> {
     def: &'a ast::TyDef,
 }
 
-impl<'a> Display for IdentTyDef<'a> {
+impl Display for IdentTyDef<'_> {
     fn fmt(&self, f: &mut Formatter<'_>) -> Result {
-        write!(
-            f,
-            "newtype {} = {}",
-            self.ident.name,
-            TyDef { def: self.def }
-        )
+        if let Some(fields) = as_struct(self.def) {
+            write!(f, "struct {} ", self.ident.name)?;
+
+            write!(f, "{{ ")?;
+            if let Some((last, most)) = fields.split_last() {
```

Comment:
Looks like this split_last() is used not to print that comma after the last element. Is this how this is typically done in Rust? My concern is that the code to represent the element as a string is repeated twice. That way changes to the visual representation will require two changes in the code. Consider looking at alternatives. There's old-fashioned approach (mutable variable to control the printing of the comma) - fast code. There's functional approach via map/join with a lambda - concise code.
Also feel free to ignore this comment here and in all other places.

Reply:
I wrote a `fmt_brace_seq` function to abstract over this logic (it was used in several places). Note that this function can't replace the similar `display_type_params` (because that needs to return a String instead of write to output) or the `fmt_tuple` (because that needs to support trailing comma for single-element tuples).